core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -1193,6 +1193,18 @@ public void getState_loadBalancerDoesNotSupportChannelState() {
   }
 
   @Test
+  public void getState_loadBalancerDoesNotSupportChannelStateThenChannelShutdown() {
+    createChannel(new FakeNameResolverFactory(false), NO_INTERCEPTOR);
+    assertEquals(ConnectivityState.IDLE, channel.getState(false));
+    helper.updatePicker(mockPicker);
+
+    channel.shutdown();
```

Comment:
Because `gotoState()` is called in `ChannelExecutor`, which catches all exceptions, this `shutdown()` will not throw even without the fix.
Perhaps we can add an `assert false` in `ChannelExecutor`'s exception handling block, to actually break test (assuming tests all have assertion enabled).